core/src/main/java/org/apache/iceberg/rest/RESTSessionCatalog.java

```diff
@@ -237,29 +199,21 @@ public void initialize(String name, Map<String, String> unresolved) {
           ResourcePaths.tokens(),
           OAuth2Properties.OAUTH2_SERVER_URI);
     }
-    String oauth2ServerUri =
-        props.getOrDefault(OAuth2Properties.OAUTH2_SERVER_URI, ResourcePaths.tokens());
+
+    this.authManager = AuthManagers.loadAuthManager(props);
```

Comment:
We should probably create the auth manager for the initial config load as a separate instance that we discard.

Reply:
See my other comment for discussion: https://github.com/apache/iceberg/pull/10753#discussion_r1808778342